src/Bridge/Doctrine/Orm/Extension/PaginationExtension.php

```diff
@@ -108,14 +115,55 @@ public function supportsResult(string $resourceClass, string $operationName = nu
     /**
      * {@inheritdoc}
      */
-    public function getResult(QueryBuilder $queryBuilder)
+    public function getResult(QueryBuilder $queryBuilder/*, string $resourceClass, string $operationName = null*/)
     {
+        $resourceClass = $operationName = null;
+
+        if (func_num_args() >= 2) {
+            $resourceClass = func_get_arg(1);
+        } else {
+            @trigger_error(sprintf('Method %s() will have a 2nd `string $resourceClass` argument in version 3.0. Not defining it is deprecated since 2.1.', __METHOD__), E_USER_DEPRECATED);
+        }
+
+        if (func_num_args() >= 3) {
+            $operationName = func_get_arg(2);
+        } else {
+            @trigger_error(sprintf('Method %s() will have a 3rd `string $operationName = null` argument in version 3.0. Not defining it is deprecated since 2.1.', __METHOD__), E_USER_DEPRECATED);
+        }
+
         $doctrineOrmPaginator = new DoctrineOrmPaginator($queryBuilder, $this->useFetchJoinCollection($queryBuilder));
         $doctrineOrmPaginator->setUseOutputWalkers($this->useOutputWalkers($queryBuilder));
 
+        $resourceMetadata = null !== $resourceClass ? $this->resourceMetadataFactory->create($resourceClass) : null;
```

Comment:
`null === $resourceClass ? null : $this->resourceMetadataFactory->create($resourceClass)` (negation not needed).